src/jit/emitxarch.cpp

```diff
@@ -10323,7 +10389,7 @@ BYTE* emitter::emitOutputRR(BYTE* dst, instrDesc* id)
         assert(IsAVXInstruction(ins) || IsSSE4Instruction(ins));
         if ((code & 0xFF00) == 0xC000)
         {
-            dst += emitOutputByte(dst, (0xC0 | regCode));
+            dst += emitOutputWord(dst, code | (regCode << 8));
```

Comment:
There's probably something I'm missing here, but how did this go from emitting a byte to emitting a word?

Reply:
From what I can tell based on the comment above, this was trying to support the smaller encoding (which isn't supported anywhere else in the emitter).

I thought I had a comment indicating I was still looking at this in particular, but can't seem to find it anymore (maybe I just forget to submit).

Reply:
@CarolEidt, I'm finishing validating locally, but it looks like this is a dead code path and is not currently hit (for `emitOutputRR`). I can only speculate (based on the comment above this if block) that it was meant to support the smaller 2-byte VEX prefix scenario, which isn't actually working.

However, for `emitOutputInstr` it special cases `IF_RRW_RRW_CNS`, that does hit the equivalent code path, and that requires it to `emitOutputWord`.

I believe the correct fix is to refactor all three cases where this particular code pattern is (`emitOutputRR`, `emitOutputRRR`, and `IF_RRW_RRW_CNS` in `emitOutputInstr`) to do the following:
```C++
// TODO-XArch-CQ: Right now support 4-byte opcode instructions only
if ((code & 0xFF00) == 0xC000)
{
    dst += emitOutputWord(dst, code | (regCode << 8));
}
else if ((code & 0xFF) == 0x00)
{
    // This case happens for SSE4/AVX instructions only
    assert(IsAVXInstruction(ins) || IsSSE4Instruction(ins));

    dst += emitOutputByte(dst, (code >> 8) & 0xFF);
    dst += emitOutputByte(dst, (0xC0 | regCode));
}
else
{
    dst += emitOutputWord(dst, code);
    dst += emitOutputByte(dst, (0xC0 | regcode));
}
```

Reply:
Changed to the above. I validated that the original code path was never hit for `emitOutputRR` and `emitOutputRRR`.

Reply:
Awesome! Thanks for checking this out and cleaning it up.